src/training_hub/algorithms/speculative_decoding.py

```
@@ -0,0 +1,1224 @@
+            "target_model_path": verifier,
+            "train_data_paths": train_data_paths,
+            "seq_length": seq_length,
+            "max_samples": max_samples,
+            "token_freq_path": token_freq_path,
+        }
+        # trust_remote_code added in speculators > 0.5.0
+        import inspect
+        if "trust_remote_code" in inspect.signature(load_and_preprocess_dataset).parameters:
+            preprocess_kwargs["trust_remote_code"] = trust_remote_code
+
+        dataset, processor = load_and_preprocess_dataset(**preprocess_kwargs)
```

Ruff (0.15.15)

[warning] Unpacked variable `processor` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)